src/decoder/mod.rs

```diff
@@ -675,6 +751,26 @@ impl<R: Read + Seek> Decoder<R> {
                 }
                 bytes
             }
+            (ColorType::Gray(32), DecodingBuffer::F32(ref mut buffer)) => {
+                reader.read_f32_into(&mut buffer[..bytes / 4])?;
+                if self.photometric_interpretation == PhotometricInterpretation::WhiteIsZero {
+                    for datum in buffer[..bytes / 4].iter_mut() {
+                        // FIXME: assumes [0, 1) range for floats
```

Comment:
That seems fine, I'm remarking this in order to remind us of opening a tracking issue to investigate how this could be validated or how a violation of this assumption might be detected.

Reply:
This is a bit complicated -- my understanding is pre-6.0 TIFF specs used `MinSampleValue` and `MaxSampleValue` as the limits that `WhiteIsZero` and `BlackIsZero` rescaled values into. By spec 6 this was dropped, and those fields became essentially meaningless. Separate `SMinSampleValue`/`SMaxSampleValue` were introduced, but these were just memoized ranges of the image and don't say anything about the sample value domain. The 6 spec just says 2^BitsPerSample, but that's because float is very much an afterthought in the spec. In practice, [0, 1) seems to be what most programs expect for float; we could check the `SMinSampleValue`/`SMaxSampleValue` (if present) to be inside this interval and error in this case.

Reply:
That's pretty much all of the information that I was missing here. If the missing parts are mostly checking for deprecated attributes and consistency checks that are seldomly relevant then this seems fine.